README.md

```diff
@@ -0,0 +1,137 @@
+            .worker_threads(2 + consensus::config::ACCUMULATOR_WORKERS_AMOUNT)
+			// Enable the time driver so we can use timeout feature in all steps execution.
+            .enable_time()
+            .build()
+            .unwrap()
+            .block_on(async { ... } )
+ ```
+
+# Build, Run and Test
+```bash
+# Run unit tests
+cargo test
+```
+
+```bash
+# Build consensus
+cargo b --release
+```
+
+```bash
+# Build and Run in-process testbed example
+cargo run --release --example testbed
+```
+
+```bash
+# Build example node
+cargo b --release --example node
+
+# Run example node
+export DUSK_WALLET_DIR="TBD"
+export DUSK_CONSENSUS_KEYS_PASS="TBD"
+
+USAGE:
+    node --bootstrap <bootstrap>... --address <public_address>  --preloaded-num <preloaded-num> --provisioner-unique-id <prov-id>  --log-level <LOG>
+
+```
+
+[^1]: A finality guarantee that is achieved through the accumulation of blocks over time, such that the probability of a block being reversed decreases exponentially as more blocks are added on top of it. This type of guarantee is in contrast to absolute finality, which is achieved when it is mathematically impossible for a block to be reversed.
+[^2]: A type of Proof-of-Stake consensus mechanism that relies on a committee of validators, rather than all validators in the network, to reach consensus on the next block. Committee-based PoS mechanisms often have faster block times and lower overhead than their non-committee counterparts, but may also be more susceptible to censorship or centralization.
+[^3]: Kadcast is a decentralized protocol used for efficient communication between nodes in a network. It is based on the Kademlia algorithm and maintains a routing table to find the best path to another node. Kadcast is commonly used in decentralized systems, such as decentralized applications (DApps) and decentralized file sharing systems, to send messages and data between nodes in a fast and reliable manner. One of the main advantages of Kadcast is its decentralized nature, which makes it resistant to censorship and other forms of interference, making it a popular choice for applications that require decentralization. Please find the whitepaper [here](https://eprint.iacr.org/2021/996). Dusk's implementation can be found [on its repo](https://github.com/dusk-network/kadcast).
+[^4]: In the Selection phase, a replica (or Provisioner in Dusk terminolody) acting as a Block Generator, proposes a block. In the First Reduction phase, a sampled committee of Provisioners vote on the validity of the block by signing a message with a BLS signature and broadcasting it. In the Second Reduction phase, a different sampled committee of Provisioners vote on the block, gather the results of the First Reduction phase, and produce an "Agreement" message. The steps keep iterating until the process is interrupted by the `Agreement` loop. This allows the group to reach a final decision about which block to agree on.
```

Comment:
A bit late to the party, but there's a typo in here. `terminolody` should be `terminology`. Something you could maybe put in your branch @fed-franz for issue #38.

I must say it was a pleasant read @goshawk-3. :) Looking forward to future additions.